pygeoapi/provider/xarray_.py

```diff
@@ -338,9 +347,10 @@ def gen_covjson(self, metadata, data, fields):
 
         return cj
 
-    def _get_coverage_properties(self):
+    def _get_coverage_properties(self, provider_def: dict):
```

Comment:
It looks like `provider_def` is now passed around in order to use `storage_crs` in `_parse_storage_crs`.

Can we set `self.storage_crs` instead in the init?